app/javascript/crayons/MultiSelectAutocomplete/MultiSelectAutocomplete.jsx

```diff
@@ -64,78 +102,215 @@ export const MultiSelectAutocomplete = ({ labelText, fetchSuggestions }) => {
-    } else {
-      // Remove inline style added to size the input
-      input.style.width = '';
-      input.focus();
+
+      // Trigger the input event to make sure suggestion UI updates correctly
+      const changeEvent = new Event('input');
+      input.dispatchEvent(changeEvent);
     }
   }, [inputPosition, editValue]);
 
+  useEffect(() => {
+    if (activeDescendentIndex !== null) {
+      const activeItem = document.querySelector('[aria-selected="true"]');
+      const { current: popover } = popoverRef;
+      if (!popover || !activeItem) return;
+
+      // Make sure that the active item is scrolled into view, if need be
+      const { offsetHeight, offsetTop } = activeItem;
+      const { offsetHeight: popoverOffsetHeight, scrollTop } = popover;
+
+      const isAbove = offsetTop < scrollTop;
+      const isBelow =
+        offsetTop + offsetHeight > scrollTop + popoverOffsetHeight;
+
+      if (isAbove) {
+        popover.scrollTo(0, offsetTop);
+      } else if (isBelow) {
+        popover.scrollTo(0, offsetTop - popoverOffsetHeight + offsetHeight);
+      }
+    }
+  }, [activeDescendentIndex]);
+
+  const selectByText = ({
+    textValue,
+    nextInputValue = '',
+    keepSelecting = true,
+  }) => {
+    const matchingSuggestion = suggestions.find(
+      (suggestion) => suggestion.name === textValue,
+    );
+    selectItem({
+      selectedItem: matchingSuggestion
+        ? matchingSuggestion
+        : { name: textValue },
+      nextInputValue,
+      keepSelecting,
+    });
+
+    // Start the next search
+    if (keepSelecting) {
+      handleAutocompleteStart();
+    }
+  };
+
   const enterEditState = (editItem, editItemIndex) => {
-    inputSizerRef.current.innerText = editItem;
+    inputSizerRef.current.innerText = editItem.name;
     deselectItem(editItem);
 
     dispatch({
       type: 'updateEditState',
-      payload: { editValue: editItem, inputPosition: editItemIndex },
+      payload: {
+        editValue: editItem.name,
+        inputPosition: editItemIndex,
+      },
     });
   };
 
-  const exitEditState = (nextInputValue = '') => {
+  const exitEditState = ({ nextInputValue = '', keepSelecting = true }) => {
+    // Reset 'edit mode' input resizing
+    inputRef.current?.style?.removeProperty('width');
     inputSizerRef.current.innerText = nextInputValue;
+
     dispatch({
       type: 'updateEditState',
       payload: {
         editValue: nextInputValue,
         inputPosition: nextInputValue === '' ? null : inputPosition + 1,
       },
     });
+
+    // Blurring away while clearing the input
+    if (!keepSelecting && nextInputValue === '') {
+      inputRef.current.value = '';
+    }
   };
 
   const resizeInputToContentSize = () => {
-    inputRef.current.style.width = `${inputSizerRef.current.clientWidth}px`;
+    const { current: input } = inputRef;
+
+    if (input) {
+      input.style.width = `${inputSizerRef.current.clientWidth}px`;
```

Comment:
nitpick (non-blocking): Prefer destructuring `current` instead of accessing it everywhere via `inputSizerRef.current`

```javascript
const { current: inputSizer } = inputSizerRef;
```

Reply:
We only reference `inputSizerRef.current` once in this function, so I figured it was neater not to unpack it into a `const`

Reply:
Cool cool. I thought I saw it multiple places. All good!